sql/ttl/ttljob: TestRowLevelTTLJobMultipleNodes failed [incorrect number of span partitions]

sql/ttl/ttljob.TestRowLevelTTLJobMultipleNodes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20776002?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20776002?buildTab=artifacts#/) on master @ [dba914a42e73cbb22ca24dcc58fc89fde8be87a6](https://github.com/cockroachdb/cockroach/commits/dba914a42e73cbb22ca24dcc58fc89fde8be87a6):


```
=== RUN   TestRowLevelTTLJobMultipleNodes/2_splits_(min_cluster_version)
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/2_splits_(min_cluster_version)
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=3 actual=2
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/2_splits_(min_cluster_version) (2.74s)
=== RUN   TestRowLevelTTLJobMultipleNodes/2_splits
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/2_splits
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=3 actual=2
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/2_splits (2.81s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobMultipleNodes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57235

sql/ttl/ttljob.TestRowLevelTTLJobMultipleNodes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20777465?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20777465?buildTab=artifacts#/) on master @ [b2158da8f70c76ccf6570689824c505270c9cd2b](https://github.com/cockroachdb/cockroach/commits/b2158da8f70c76ccf6570689824c505270c9cd2b):


```
=== RUN   TestRowLevelTTLJobMultipleNodes
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/be120b58d19356f96778e6065320d34c/logTestRowLevelTTLJobMultipleNodes2017946710
    test_log_scope.go:82: use -show-logs to present logs inline
    ttljob_test.go:748: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be120b58d19356f96778e6065320d34c/logTestRowLevelTTLJobMultipleNodes2017946710
--- FAIL: TestRowLevelTTLJobMultipleNodes (15.15s)
=== RUN   TestRowLevelTTLJobMultipleNodes/1_split_(min_cluster_version)
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/1_split_(min_cluster_version)
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=2 actual=1
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/1_split_(min_cluster_version) (2.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobMultipleNodes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/ttl/ttljob.TestRowLevelTTLJobMultipleNodes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20778518?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20778518?buildTab=artifacts#/) on master @ [2b2cfc43744736e47310a3ec7ccef049650da1c7](https://github.com/cockroachdb/cockroach/commits/2b2cfc43744736e47310a3ec7ccef049650da1c7):


```
=== RUN   TestRowLevelTTLJobMultipleNodes/1_split
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/1_split
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=2 actual=1
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/1_split (1.99s)
=== RUN   TestRowLevelTTLJobMultipleNodes/1_split_(min_cluster_version)
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/1_split_(min_cluster_version)
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=2 actual=1
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/1_split_(min_cluster_version) (2.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobMultipleNodes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/ttl/ttljob/ttljob_test.go
+++ b/pkg/sql/ttl/ttljob/ttljob_test.go
@@ -128,6 +128,8 @@ func newRowLevelTTLTestJobTestHelper(
 			Settings:          makeSettings(),
 			Knobs:             baseTestingKnobs,
 			InsecureWebAccess: true,
+			// TODO(server): re-enable DRPC once flakiness is addressed, see #158387.
+			DefaultDRPCOption: base.TestDRPCDisabled,
 		},
 	})
 	th.testCluster = testCluster
@@ -593,7 +595,6 @@ INSERT INTO t (id, crdb_internal_expiration) VALUES (1, now() - '1 month'), (2, 
 func TestRowLevelTTLJobMultipleNodes(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.WithIssue(t, 158317)
 
 	skip.UnderRace(t, "the test times out under race")
 

